src/components/board/Stack.vue

```diff
@@ -43,9 +43,12 @@
 				</form>
 			</transition>
 			<Actions v-if="canManage && !isArchived" :force-menu="true">
-				<ActionButton icon="icon-archive" @click="modalArchivAllCardsShow=true">
+				<ActionButton v-if="!showArchived" icon="icon-archive" @click="modalArchivAllCardsShow=true">
 					{{ t('deck', 'Archive all cards') }}
 				</ActionButton>
+				<ActionButton v-if="showArchived" icon="icon-archive" @click="setArchivedToAllCardsFromStack(stack, false)">
```

Comment:
Unarchiving is also a long running operation so it would be good to also show the modal with the progress are here, as it is done for the archive all action.